src/coreclr/src/jit/importer.cpp

```diff
@@ -756,51 +756,36 @@ void Compiler::impInsertTreeBefore(GenTree* tree, IL_OFFSETX offset, Statement*
  *  curLevel is the stack level for which the spill to the temp is being done.
  */
 
-void Compiler::impAssignTempGen(unsigned    tmp,
-                                GenTree*    val,
-                                unsigned    curLevel,
-                                Statement** pAfterStmt, /* = NULL */
-                                IL_OFFSETX  ilOffset,   /* = BAD_IL_OFFSET */
-                                BasicBlock* block       /* = NULL */
-                                )
+void Compiler::impAssignTempGenTree(unsigned tmp, GenTree* val, unsigned curLevel, IL_OFFSETX offset)
```

Comment:
If you want to change this, would be better to rename to something that includes "append" in the name? It's the main difference between this and gtNewTempAssign.

Reply:
I agree, but right now I am not sure that I will do a reviewable/mergeable PR out of this.

I want to get rid of some pessimisations for calls that return structs. They hide somewhere down that rabbit hole of `impAssignStruct`, so that PR helps me to understand what really happens there.